### DISCUSSION TITLE
Data not being uploaded to realtime database with @react-native-firebase/database

### DISCUSSION THREAD

**Discussion by g-mahendra** [invertase/react-native-firebase#4215] [Q&A] [answered]
I am using @react-native-firebase/database on android for an app to store data. At first day, I was able to store and retrieve the data from firebase but one day after that day, nothing seems to work. When I create a reference to an existing object on database, to retrieve data, it does not retrieve the data even if I store the data manually in firebase console in database. When I use that same reference to store data, it does not store that data to firebase console but that data can be retrieved at that time. It is temporary. That means, when I store some data to some reference, it gets stored temporarily and can be retrieved as long as the app is not restarted. After restarting the app, the data is gone. It never shows up in firebase console. Suggest any fix.
Tried creating a new rn project with a fresh firebase project but results are the same.
Thanks in advance.
Here are my codes to store and retrieve data.
The firebase cloud storage works just fine.
react-native version 0.63.2
@react-native-firebase/database version 7.5.1
        
        //code to store data
        import * as React from 'react';
        import {
          Text,
          View,
          StyleSheet,
          TextInput,
          TouchableOpacity,
        } from 'react-native';
        import {Picker} from '@react-native-community/picker';
        import Icon from 'react-native-vector-icons/Feather';
        import database from '@react-native-firebase/database';
        class UploadScreen extends React.Component {
          state = {
            sem: null,
            branch: null,
            subject: null,
            year: null,
            pic: 'mg',
          };
          uploadObject() {
            const databaseRef = database().ref('users');
            console.log(databaseRef);
            const sendObj = databaseRef.push().set({
              sem: this.state.sem,
              branch: this.state.branch,
              subject: this.state.subject,
              year: this.state.year,
            });
          }

        //code to retrieve the data
        import * as React from 'react';
        import {
          Text,
          View,
          StyleSheet,
          TextInput,
          TouchableOpacity,
        } from 'react-native';
        import {Picker} from '@react-native-community/picker';
        import database from '@react-native-firebase/database';
        import Icon from 'react-native-vector-icons/Feather';
        class HomeScreen extends React.Component {
          state = {
            sem: null,
            branch: null,
            subject: null,
            year: null,
            collected: [],
          };

          componentDidMount() {
            const myref = database().ref('users');
            console.log(myref);
            myref.on('value', (snapshot) => {
              console.log(snapshot.val());
            });
          }

**Comment by mikehardy** [comment, marked as answer]
You should open a real issue, and follow the issue template, along with this App.js
Prior to doing that though, you should *definitely* add error handling to your code. In no place are you doing try/catch and checking for errors.
